src/chocolatey.tests.integration/scenarios/InfoScenarios.cs

```diff
@@ -0,0 +1,315 @@
+        public class when_searching_for_exact_package_with_verbose_output : ScenariosBase
+        {
+            public override void Context()
+            {
+                base.Context();
+
+                Configuration.PackageNames = Configuration.Input = "installpackage";
+                Configuration.Sources = "PackageOutput";
+                Scenario.add_packages_to_source_location(Configuration, "installpackage.*" + NuGetConstants.PackageExtension);
+            }
+
+            [Fact]
+            public void should_show_only_one_result()
+            {
+                Results.Count.ShouldEqual(1, "Expected 1 package to be returned!");
+            }
+
+            [Fact]
+            public void should_set_exit_code_to_zero()
+            {
+                Results[0].ExitCode.ShouldEqual(0);
+            }
+
+            [Fact]
+            public void should_not_be_reported_as_inconclusive()
+            {
+                Results[0].Inconclusive.ShouldBeFalse();
+            }
+
+            [Fact]
+            public void should_report_expected_name()
+            {
+                Results[0].Name.ShouldEqual("installpackage");
+            }
+
+            [Fact]
+            public void should_set_source_to_expected_value()
+            {
+                Results[0].Source.ShouldEqual("PackageOutput");
+            }
+
+            [Fact]
+            public void should_set_expected_version()
+            {
+                Results[0].Version.ShouldEqual("1.0.0");
+            }
+        }
+
+        [Categories.SourcePriority]
+        public class when_searching_for_a_package_in_a_priority_source : ScenariosBase
+        {
+            public override void Context()
+            {
+                base.Context();
+
+                Configuration.PackageNames = Configuration.Input = "test-package";
+                Configuration.Sources = Scenario.add_packages_to_priority_source_location(Configuration, "test-package.*" + NuGetConstants.PackageExtension, priority: 1);
+            }
+
+            [Fact]
+            public void should_show_only_one_result()
+            {
+                Results.Count.ShouldEqual(1, "Expected 1 package to be returned!");
+            }
+
+            [Fact]
+            public void should_set_exit_code_to_zero()
+            {
+                Results[0].ExitCode.ShouldEqual(0);
+            }
+
+            [Fact]
+            public void should_not_be_reported_as_inconclusive()
+            {
+                Results[0].Inconclusive.ShouldBeFalse();
+            }
+
+            [Fact]
+            public void should_report_expected_name()
+            {
+                Results[0].Name.ShouldEqual("test-package");
+            }
+
+            [Fact]
+            public void should_set_source_to_expected_value()
+            {
+                var expectedSource = "file:///" + Path.Combine(

```

Comment:
Would this work on non-Windows platforms?

Reply:
At the moment this is a moot point, as these tests aren't running and won't be running until we start working on fixing the source priority itself.

There may always be changes depending on the implementation, so supporting non-windows platforms would need to be looked into at that point.